gradle-plugin/kubernetes/src/main/java/org/eclipse/jkube/gradle/plugin/task/KubernetesApplyTask.java

```diff
@@ -28,6 +45,60 @@ public KubernetesApplyTask(Class<? extends KubernetesExtension> extensionClass)
 
   @Override
   public void run() {
-    throw new UnsupportedOperationException("To be implemented");
+    try (KubernetesClient kubernetes = jKubeServiceHub.getClient()) {
+      applyService = jKubeServiceHub.getApplyService();
+
+      final File manifest = kubernetesExtension.getManifest(kitLogger, kubernetes, javaProject);
+      if (!manifest.exists() || !manifest.isFile()) {
+        if (kubernetesExtension.getFailOnNoKubernetesJsonOrDefault()) {
+          throw new IllegalStateException("No such generated manifest file: " + manifest);
+        } else {
+          kitLogger.warn("No such generated manifest file %s for this project so ignoring", manifest);
+          return;
+        }
+      }
+
+      URL masterUrl = kubernetes.getMasterUrl();
+      KubernetesResourceUtil.validateKubernetesMasterUrl(masterUrl);
+      List<HasMetadata> entities = KubernetesHelper.loadResources(manifest);
+      kitLogger.info("Using %s at %s in namespace %s with manifest %s ", OpenshiftHelper.isOpenShift(kubernetes) ? "OpenShift" : "Kubernetes", masterUrl, clusterAccess.getNamespace(), manifest);
+
+      configureApplyService();
+
+      // Apply rest of the entities present in manifest
+      applyEntities(manifest.getName(), entities);
+      kitLogger.info("HINT: Use the command `%s get pods -w` to watch your pods start up", clusterAccess.isOpenShift() ? "oc" : "kubectl");
+    } catch (KubernetesClientException e) {
+      KubernetesResourceUtil.handleKubernetesClientException(e, kitLogger);
+    } catch (IOException ioException) {
+      kitLogger.error("Error in loading Kubernetes Manifests ", ioException);
+      throw new IllegalStateException(ioException);
+    } catch (InterruptedException interruptedException) {
+      Thread.currentThread().interrupt();
+      throw new IllegalStateException(interruptedException.getMessage(), interruptedException);
+    }
+  }
+
+  private void applyEntities(String fileName, final Collection<HasMetadata> entities) throws InterruptedException {
+    KitLogger serviceLogger = createLogger(" [SVC] ");
+    applyService.applyEntities(fileName, entities, serviceLogger, kubernetesExtension.getServiceUrlWaitTimeSecondsOrDefault());
+  }
+
+  protected void configureApplyService() {
+    applyService.setAllowCreate(kubernetesExtension.getCreateNewResourcesOrDefault());
+    applyService.setServicesOnlyMode(kubernetesExtension.getServicesOnlyOrDefault());
+    applyService.setIgnoreServiceMode(kubernetesExtension.getIgnoreServicesOrDefault());
+    applyService.setLogJsonDir(kubernetesExtension.getJsonLogDirOrDefault(javaProject));
+    applyService.setBasedir(javaProject.getBaseDirectory());
+    applyService.setIgnoreRunningOAuthClients(false);
+    applyService.setProcessTemplatesLocally(true);
```

Comment:
Why are some of these variables hardcoded and not retrieved from the KubernetesExtension?

Reply:
In ApplyMojo we have this method to disable OpenShift features if the platform mode is not OpenShift:

https://github.com/eclipse/jkube/blob/a2a0934abf16fff1bcc60702d891a8f08fa76590/kubernetes-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/build/ApplyMojo.java#L219-L223

I made this inline to disable these in case of KubernetesApplyTask. I was planning to override this method to enable them again in case of OpenShiftApplyTask

For `applyService.setRollingUpgradePreserveScale(false);` I saw there was a fixed value coming from a method which was not getting overridden from anywhere. So I inlined this as well.

https://github.com/eclipse/jkube/blob/a2a0934abf16fff1bcc60702d891a8f08fa76590/kubernetes-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/build/ApplyMojo.java#L208-L210

Reply:
So for processTemplatesLocally, that's a configurable value. The right approach would be to declare this in KubernetesExtension (we already have it in OpenShiftExtension), and then use the `getProcessTemplatesLocallyOrDefault` to retrieve the default `true`.
`getProcessTemplatesLocallyOrDefault` is then overridden in OpenShiftExtension and false is returned instead.



For `applyService.setRollingUpgradePreserveScale(false)` we should dig deeper. This means that the value is always false in the apply service configuration, probably this had some evolution in FMP.
Now might be the right time to move this as a configuration parameter -> KubernetesExtension, and allow it to be configured.